src/main/java/net/mcreator/workspace/Workspace.java

```diff
@@ -488,6 +490,8 @@ public static Workspace createWorkspace(File workspaceFile, WorkspaceSettings wo
 
 	public void reloadFromFS() {
 		String workspace_string = FileIO.readFileToString(fileManager.getWorkspaceFile());
+		workspace_string = workspace_string.replaceAll("\"type\": \"mob\",", "\"type\": \"livingentity\",")

```

Comment:
You are checking this in deserializer so you don't need this here too (I think)

Reply:
The code here will change the code in the .mcreator while the code in GeneratableElement will change it for the element files (worspaceFolder/elements).

Reply:
I see. But replacing like this is not ok, JSON can have different formatting. Check how adapter is registered for soundelements here: https://github.com/MCreator/MCreator/blob/064a81aa50bc3551951d93dcba9e877dc6481fa2/src/main/java/net/mcreator/workspace/elements/SoundElement.java#L89

Make a similar adapter for modelement to refactor the type cleanly with json parsing, not by replacing texts

Reply:
This was not resolved

Reply:
What's wrong in the commit? https://github.com/MCreator/MCreator/pull/1400/commits/347a7f391c24a67e92324e0411359a179b68f466